src/components/PostCommentForm.jsx

```diff
@@ -0,0 +1,62 @@
+import { UserContext } from "../contexts/UserContext"
+import "../styles/PostCommentForm.css"
+import { useContext, useState } from "react"
+import { postComment } from "../utils/fetches"
+import { useParams } from "react-router"
+import CommentCard from "./CommentCard"
+
+const PostCommentForm = ({ addingComment, setAddingComment }) => {
+  const { user, setUser } = useContext(UserContext)
+  const [commentValue, setCommentValue] = useState("")
+  const [fakeComment, setFakeComment] = useState({})
+  const [commentSuccessful, setCommentSuccessful] = useState(true)
+  const { article_id } = useParams()
+
+  function handleChange(e) {
+    setCommentValue(e.target.value)
+  }
+
+  function handleSubmit(e) {
+    e.preventDefault()
+    postComment(article_id, commentValue, user.username)
+      .then((data) => {
+        setFakeComment({
+          author: user.username,
+          created_at: Date.now(),
+          body: commentValue,
+          votes: 0,
+        })
+        setCommentValue("")
+        setAddingComment(true)
+        setCommentSuccessful(true)
+      })
+      .catch((err) => {
+        setCommentSuccessful(false)
+      })
+  }
```

Comment:
You don't necessarily HAVE to optimistic render a comment post - just watch out because this might cause some weirdness down the line when it comes to voting on comments, and a user tries to vote on an optimistically rendered comment. Or as you've called it, a fakeComment